stdlib/public/SwiftShims/RefCount.h

```diff
@@ -128,6 +140,25 @@ class StrongRefCount {
     }
   }
 
+  bool tryIncrementAndPinNonAtomic() {
+    uint32_t oldval = __atomic_load_n(&refCount, __ATOMIC_RELAXED);
+    while (true) {
+      // If the flag is already set, just fail.
+      if (oldval & RC_PINNED_FLAG) {
+        return false;
+      }
+
+      // Try to simultaneously set the flag and increment the reference count.
+      uint32_t newval = oldval + (RC_PINNED_FLAG + RC_ONE);
+      // Use store __ATOMIC_RELAXED here?
+      if (refCount != oldval)
```

Comment:
We don't need a CAS loop here as far as I can see, since the operation is nonatomic. We should be able to update the flags and store.